### PR TITLE
[Docs] Update readme for recommending the nightly build

### PR DESCRIPTION
**Pull request by Michaelvll**
Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,13 +55,13 @@ SkyPilot **cuts your cloud costs**:
 
 SkyPilot supports your existing GPU, TPU, and CPU workloads, with no code changes.
 
-Install with pip:
+Install with pip (we recommend using the nightly build for the latest features/updates) or [from source](https://skypilot.readthedocs.io/en/latest/getting-started/installation.html):
```
Comment (concretevitamin) [inline README.md:58]:
```suggestion
Install with pip (we recommend the nightly build for the latest features or [from source](https://skypilot.readthedocs.io/en/latest/getting-started/installation.html)):
```

File: README.md [hunk 2]
```diff
@@ -55,13 +55,13 @@ SkyPilot **cuts your cloud costs**:
...
 ```bash
```
Comment (concretevitamin) [inline README.md:59]: Wdyta `aws,gcp,azure,oci,lambda,runpod,ibm,scp,kubernetes` everywhere?

**Review by concretevitamin — APPROVED** [review]
Thanks @Michaelvll!

File: README.md (hunk 2 above)
Comment (Michaelvll) [reply README.md:59]: I changed it to `aws,gcp,azure,kubernetes,oci,lambda,runpod,ibm,scp`. Wdyt?

Comment (Michaelvll) [reply README.md:59]: Sounds good. Reordered. Thanks!

**Pull request merged**